packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2232,6 +2232,38 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
       return;
     assert(() {
       if (_needsCompositingBitsUpdate) {
+        final RenderObject parent = this.parent as RenderObject;
+        if (parent is RenderObject) {
+          bool isChildOfParent = false;
+          parent.visitChildren((RenderObject child) {
+            assert(!isChildOfParent);
```

Comment:
This assert can fail if the parent visits multiple children - it can be removed.